src/DependencyInjection/HttplugExtension.php

```diff
@@ -203,10 +203,22 @@ private function configurePluginByName($name, Definition $definition, array $con
         switch ($name) {
             case 'cache':
                 $options = $config['config'];
+
                 if (!empty($options['cache_key_generator'])) {
                     $options['cache_key_generator'] = new Reference($options['cache_key_generator']);
                 }
 
+                if (!empty($options['cache_listeners'])) {
+                    foreach ($options['cache_listeners'] as $i => $listener) {
+                        if (!empty($listener)) {
+                            $options['cache_listeners'][$i] = new Definition($listener);
```

Comment:
this is assuming that the listener classes have empty constructors. i think we should define them to be service names, and then we can create new `Reference` here instead.

because of bundle loading order, we can't assume that the service already exists at this spot. the user will get an error later when a referenced service does not exist.

theoretically we could do a compiler pass (in a step where the definitions are supposed to all exist) to check the services for implementing the interface, but i don't think we need to go to that effort.

Reply:
Switched form list of classes to list of service ids.

Removed the checks for correct implementation of given interface.